### PR TITLE
Siva-Khalin/Appeals 43239

### PR DESCRIPTION
**Pull request by Dev-KRedd**
Resolves [APPEALS-43239](https://jira.devops.va.gov/browse/APPEALS-43239)

# Description
- Created a new test `Correspondence Admin` Page where user can generate correspondence for the veterans by entering veteran file numbers and the number of correspondence needed.
- Followed this [Mock](https://www.figma.com/proto/uWc8NMuvCrNMQF5TEUDbAM/Correspondence-Admin-Page-(Appeals-43239)?page-id=0%3A1&node-id=1-1642&viewport=586%2C897%2C0.25&t=A58Nw8SlPts76dlw-1&scaling=scale-down&starting-point-node-id=1%3A1642&content-scaling=fixed) to create this page

## Acceptance Criteria
- [X] Code compiles correctly

## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below

Navigate to the Caseflow Admin page
Switch User to `Admin` or `Bva` or `Inbound Ops Team Supervisor`
Find `Correspondence Admin` link under `Queue` Tab, Click the same
And navigate to Correspondence Admin Page
Enter Veteran File Numbers and Count to Generate Correspondence.

# Frontend
## User Facing Changes
image

image

### Test Coverage
Did you include any test coverage for your code? Check below:
- [X] RSpec
- [X] Jest

### Code Climate
- [X] No new code climate issues added

### Error Handling
- [X] Are errors being caught and handled gracefully?

### CODE REVIEW

File: app/controllers/test/correspondence_controller.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  before_action :verify_feature_toggle, only: [:index]
+  def index
+    render_access_error unless verify_access && access_allowed?
+  end
+
+  def generate_correspondence
+    nums = correspondence_params[:file_numbers].split(",").map(&:strip).reject(&:empty?)
+    invalid_nums = invalid_file_numbers(nums)
+
+    valid_file_nums = "78907"
```
Comment (cacevesva) [inline app/controllers/test/correspondence_controller.rb:16]: What does this line do?

Comment (cacevesva) [reply app/controllers/test/correspondence_controller.rb:16]: Never mind I see what is going now.

File: app/controllers/test/correspondence_controller.rb [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        valid_file_nums << file_number
+      else
+        invalid_file_nums << file_number
+      end
+    end
+
+    { valid: valid_file_nums, invalid: invalid_file_nums }
+  end
+
+  def connect_corr_with_vet(valid_veterans, count)
```
Comment (cacevesva) [inline app/controllers/test/correspondence_controller.rb:94]: @psivakeshva We are on Rails 6 do the need to be enabled?

File: client/test/app/test/TestCorrespondence.test.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    render(<TestCorrespondence {...props} />);
+    const textarea = screen.getByRole('textbox');
+
+    fireEvent.change(textarea, { target: { value: '1,2,3,4,5,6,7,8,9,10,11' } });
+    expect(textarea.value).toBe('1,2,3,4,5,6,7,8,9,10');
+  });
+
+  it('handles correspondence count input', () => {
+    render(<TestCorrespondence {...props} />);
+    const numberField = screen.getByRole('spinbutton');
```
Comment (cacevesva) [inline client/test/app/test/TestCorrespondence.test.js:67]: Is this suppose to be 1 word or two words?

Comment (psivakeshva) [reply client/test/app/test/TestCorrespondence.test.js:67]: 1 word

File: app/controllers/test/correspondence_controller.rb (hunk 2 above)
Comment (psivakeshva) [reply app/controllers/test/correspondence_controller.rb:94]: No we don't need that. I removed this piece of commented code.

**Review by cacevesva — APPROVED** [review]

File: Gemfile.lock
Comment (cacevesva) [inline Gemfile.lock:1]: Why are we adding gemfile.lock this needs to be removed?
@psivakeshva

**Pull request merged**